src/components/Query.ts

```diff
@@ -41,18 +44,27 @@ export class Engine {
     newNodeFn: (
       name: string,
       outputChannel: OutputChannel,
-      iConn: InfluxDBConnection
-    ) => INode
+      iConn: InfluxDBConnection,
+      parent?: string
+    ) => INode,
+    v2mode?: boolean,
+    pp?: string
   ): Promise<INode[]> {
     this.outputChannel.show();
     this.outputChannel.appendLine(`${now()} - ${msg}`);
-    let result: Result = await APIRequest.Query(conn, query);
+    let result: Result;
+    if (conn.version !== InfluxConnectionVersion.V1 || v2mode) {
+      result = await APIRequest.Query(conn, query);
+    } else {
+      result = await APIRequest.QueryV1(conn, query, pp as string);
```

Comment:
using this type of casting breaks down the type safety and should be considered a code smell